dbt_copilot_helper/commands/application.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+import time
+from datetime import datetime
+from datetime import timedelta
+
+import click
+
+from dbt_copilot_helper.utils.aws import get_aws_session_or_abort
+from dbt_copilot_helper.utils.click import ClickDocOptGroup
+from dbt_copilot_helper.utils.versioning import (
+    check_copilot_helper_version_needs_update,
+)
+
+
+def get_query_results(env, app, profile, query_string):
+
+    timeout = 15
+    label_text = click.style("Waiting for results:", fg="yellow")
+    fill_char = click.style("#", fg="yellow")
+    empty_char = click.style("-", fg="yellow", dim=True)
+
+    project_session = get_aws_session_or_abort(profile)
+
+    click.secho(
+        f"Showing status for app {app} in aws account {profile}",
+        fg="green",
+    )
+    logs_client = project_session.client("logs")
+    ecs_client = project_session.client("ecs")
+
+    response = ecs_client.list_clusters()
+
+    # Find clusters log-group
+    for cluster in response["clusterArns"]:
+        cluster_name = cluster.split("/")[-1]
+        APP = cluster_name.split("-")[0]
+        ENV = cluster_name.split("-")[1]
+        if app == APP and env == ENV:
+            log_group_name = f"/aws/ecs/containerinsights/{cluster_name}/performance"
+            break
+
+    # Container stats are 5 mins behind realtime.
+    date_time = datetime.now()
+    end_time = int(date_time.timestamp())
+    start_time = int((date_time - timedelta(minutes=5)).timestamp())
+    end_time = int((date_time - timedelta(minutes=4)).timestamp())
+
+    click.echo(
+        click.style("Date & Time:  ", fg="cyan")
+        + click.style(f"{datetime.utcfromtimestamp(start_time)}", fg="cyan", bold=True)
+    )
+
+    # First you need to create a log query, then retrieve the log query.
+    cpu_response_id = logs_client.start_query(
+        logGroupName=log_group_name,
+        startTime=start_time,
+        endTime=end_time,
+        queryString=query_string,
+    )
+
+    # Need to wait a few seconds for the query to be available, up to timeout value.
+    with click.progressbar(label=label_text, length=timeout, show_eta=False, show_percent=False, fill_char=fill_char, empty_char=empty_char) as progress_bar:
+        while len(logs_client.get_query_results(queryId=cpu_response_id["queryId"])['results']) == 0 and timeout > 0:
+            time.sleep(1)
+            progress_bar.update(1)
+            timeout = timeout - 1
+
+    if timeout == 0:
+        click.secho("Timeout threshold breached, no results returned.\nExiting...", fg="red")
```

Comment:
```suggestion
        click.secho(f"Timeout threshold breached, no results returned in {timeout} seconds.\nExiting...", fg="red")
```

Reply:
I was toying with this but cant do that, as timeout is counting backwards.  So that value is 0 when it get to that line.
Line 67: timeout = timeout - 1

Reply:
Ah, yes, I normally run the other way round incrementing an `elapsed` variable up from `0` so the `timeout` remains unchanged.

Reply:
ok switching to elapsed